integration/helpers/x509.go

```diff
@@ -0,0 +1,71 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package helpers
+
+import (
+	"crypto/x509"
+	"sync"
+	"testing"
+	_ "unsafe"
+)
+
+//go:linkname x509_systemRootsMu crypto/x509.systemRootsMu
+//go:linkname x509_systemRoots crypto/x509.systemRoots
```

Comment:
Can we at least do this in a _test.go file to be sure it doesn't ever affect a production build?

Reply:
The issue with that is that other packages' test files don't exist when building tests.

Should I move this back to `tools/tsh/common/proxy_test.go` for now?

Reply:
I figured out that with `//go:debug x509usefallbackroots=1` (which is far more supported) we get to replace the system cert pool without using linkname hacks or changing all the test running environments to add the matching GODEBUG envvar (b986a46)